pkg/repository/task_list_sql.go

```diff
@@ -0,0 +1,205 @@
+import (
+	"database/sql"
+	"fmt"
+	"strconv"
+	"time"
+
+	tasks "github.com/Extrarius/tido"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	query string
+)
+
+type TaskListSQL struct {
+	db *sql.DB
+}
+
+func NewTaskListSQL(db *sql.DB) *TaskListSQL {
+	return &TaskListSQL{db: db}
+}
+
+// Реализация метода Create
+func (r *TaskListSQL) Create(tasks tasks.Tasks) (string, error) {
+
+	// Сохранение задачи в базу данных
+	query = "INSERT INTO scheduler (date, title, comment, repeat) VALUES (?, ?, ?, ?)"
+	res, err := r.db.Exec(query, tasks.Date, tasks.Title, tasks.Comment, tasks.Repeat)
+	if err != nil {
+		logrus.Println("Ошибка при записи в базу данных", err)
+		return "", err
+	}
+
+	// Получение ID созданной записи
+	id, err := res.LastInsertId()
+	if err != nil {
+		logrus.Println("Ошибка при получении id записи в базе данных", err)
+		return "", err
+	}
+	idStr := strconv.Itoa(int(id))
+
+	return idStr, err
+}
+
+func (r *TaskListSQL) GetAll(search string) ([]tasks.Tasks, error) {
+	var (
+		lists []tasks.Tasks
+		rows  *sql.Rows
+		err   error
+		date  time.Time
+	)
+
+	if search == "" {
+		now := time.Now().Format(format)
+		query = fmt.Sprintf("SELECT * FROM scheduler WHERE date >= '%s' ORDER BY date;", now)
+
+		logrus.Infof("Выполнение запроса: %s", query)
+
+		rows, err = r.db.Query(query)
+		if err != nil {
+			logrus.Errorf("Ошибка при выполнении запроса: %v", err)
+			return nil, err
+		}
+	} else {
+		// Проверка, является ли строка поиска датой в формате 02.01.2006
+		if date, err = time.Parse("02.01.2006", search); err == nil {
+			dateS := date.Format(format)
+			query = "SELECT * FROM scheduler WHERE date = ? ORDER BY date"
+			//date хранится в 20060102, а search передается 02.01.2006
+			rows, err = r.db.Query(query, dateS)
+			if err != nil {
+				logrus.Errorf("Ошибка при выполнении запроса: %v", err)
+				return nil, err
+			}
+		} else {
+			query = "SELECT * FROM scheduler WHERE title LIKE ? OR Comment LIKE ? ORDER BY date"
+			searchPattern := "%" + search + "%"
+			rows, err = r.db.Query(query, searchPattern, searchPattern)
+			if err != nil {
+				logrus.Errorf("Ошибка при выполнении запроса: %v", err)
+				return nil, err
+			}
+		}
+	}
+	defer func() {
+		err := rows.Close()
+		if err != nil {
+			logrus.Errorf("Ошибка при закрытии запроса rows: %v", err)
+		}
+	}()
+
+	for rows.Next() {
+		var task tasks.Tasks
+		if err := rows.Scan(&task.ID, &task.Date, &task.Title, &task.Comment, &task.Repeat); err != nil {
+			logrus.Errorf("Ошибка при сканировании строки: %v", err)
+			return nil, err
+		}
+		lists = append(lists, task)
+	}
+
+	if lists == nil {
+		logrus.Infof("Список задач пуст.")
+		lists = make([]tasks.Tasks, 0)
+	} else {
+		logrus.Infof("Найдено %d задач.", len(lists))
+	}
+
+	if err := rows.Err(); err != nil {
+		logrus.Errorf("Ошибка результата запроса строк: %v", err)
+		return nil, err
+	}
+
+	return lists, nil
+}
+
+func (r *TaskListSQL) GetById(taskId string, tasks tasks.Tasks) (tasks.Tasks, error) {

```

Comment:
На входе только айди должен быть
```suggestion
func (r *TaskListSQL) GetById(taskId string) (tasks.Tasks, error) {
```